LiteCore/Database/TreeDocument.cc

```diff
@@ -640,14 +644,51 @@ namespace litecore {
             return true;
         }
 
+        static bool hasEncryptables(slice body, SharedKeys* sk) {
+#ifndef COUCHBASE_ENTERPRISE
+            return false;
+#else
+            const Value* v = Value::fromTrustedData(body);
+            if (v == nullptr) {
+                return false;
+            }
+
+            struct AuxDoc : public Doc {
```

Comment:
I don't see a need to subclass Doc; you're just passing different parameters to the constructor.

Reply:
The reason I subclass it is because it is in fleece::impl and here fleece::impl::Doc has destructor being protected.

Reply:
That's because Doc is refcounted. You never allocate a RefCounted object on the stack, only on the heap, and use a Retained value to manage the reference.

The protected destructor is explicitly meant to keep people from trying to do what you were doing.